### PR TITLE
CI for Prefect owned exporter

### PR DESCRIPTION
**Pull request by jimid27**
- Relates: https://github.com/PrefectHQ/prefect-helm/issues/263
- Creates Dockerhub releases for prometheus exporter for prefect
- Adds Platform team to CODEOWNERS
- Relevent Dockerhub repo: https://hub.docker.com/repository/docker/prefecthq/prometheus-prefect-exporter/general

### CODE REVIEW

File: .github/workflows/release_build_deploy.yaml [hunk 1]
```diff
@@ -1,58 +1,32 @@
...
-    branches:
-      - main
     tags:
-      - 'v*'
+      - '*'
 
 env:
-  USER: devopsiaci
+  USER: prefecthq
   REPO: prometheus-prefect-exporter
```
Comment (jamiezieziula) [inline .github/workflows/release_build_deploy.yaml:10]: i don't think we need these definitions - we can just hardcode the org/repo combo below

File: .github/workflows/release_build_deploy.yaml [hunk 2]
```diff
@@ -1,58 +1,32 @@
 name: "[prometheus-prefect-exporter] Create release, build and push"
```
Comment (jamiezieziula) [inline .github/workflows/release_build_deploy.yaml:1]:
```suggestion
name: "Release Helm Chart"
```

File: .github/workflows/release_build_deploy.yaml [hunk 3]
```diff
@@ -1,58 +1,32 @@
...
 
 on:
```
Comment (jamiezieziula) [inline .github/workflows/release_build_deploy.yaml:3]:
```suggestion
"on":
```

File: .github/workflows/release_build_deploy.yaml (hunk 2 above)
Comment (jamiezieziula) [inline .github/workflows/release_build_deploy.yaml:1]:
```suggestion
name: "Build and publish prometheus-prefect-exporter image"
```

File: .github/workflows/release_build_deploy.yaml [hunk 4]
```diff
@@ -1,58 +1,32 @@
...
     tags:
-      - 'v*'
+      - '*'
 
 env:
-  USER: devopsiaci
+  USER: prefecthq
   REPO: prometheus-prefect-exporter
 
 jobs:
```
Comment (jamiezieziula) [inline .github/workflows/release_build_deploy.yaml:12]:
```suggestion
# Do not grant jobs any permissions by default
permissions: {}

jobs:
```

File: .github/workflows/release_build_deploy.yaml [hunk 5]
```diff
@@ -1,58 +1,32 @@
...
-  create-tag-release:
+  build-push-exporter:
```
Comment (jamiezieziula) [inline .github/workflows/release_build_deploy.yaml:13]:
```suggestion
  build_push_exporter:
```

File: .github/workflows/release_build_deploy.yaml [hunk 6]
```diff
@@ -1,58 +1,32 @@
...
-
-  build-push:
-    runs-on: ubuntu-latest
-    if: github.ref_name != 'main'
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v3
-
       - name: Set up QEMU
         uses: docker/setup-qemu-action@v3
```
Comment (jamiezieziula) [inline .github/workflows/release_build_deploy.yaml:17]: what does this do?

File: .github/workflows/release_build_deploy.yaml [hunk 7]
```diff
@@ -1,58 +1,32 @@
...
 
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v3
```
Comment (jamiezieziula) [inline .github/workflows/release_build_deploy.yaml:20]: this make sense that we need this, but I don't see if referenced anywhere downstream?

File: .github/workflows/release_build_deploy.yaml [hunk 8]
```diff
@@ -1,58 +1,32 @@
 name: "[prometheus-prefect-exporter] Create release, build and push"
 
 on:
-  workflow_dispatch:
   push:
-    branches:
-      - main
     tags:
-      - 'v*'
+      - '*'
```
[the remark this answers is not in the document]
Comment (jimid27) [reply .github/workflows/release_build_deploy.yaml:6]: Torn on this.  I think yes, though we likely will not be releasing this often like our helm charts

[the remark this answers is not in the document]
Comment (jamiezieziula) [reply .github/workflows/release_build_deploy.yaml:6]: yeah i said this before i realized this was just a docker image. i think SemVer makes sense

File: .github/workflows/release_build_deploy.yaml (hunk 6 above)
Comment (jimid27) [reply .github/workflows/release_build_deploy.yaml:19]: allows us to build and push with diff architectures

File: .github/workflows/release_build_deploy.yaml (hunk 7 above)
Comment (jimid27) [reply .github/workflows/release_build_deploy.yaml:22]: Same as above - we need to build in multiple architectures

File: .github/workflows/release_build_deploy.yaml (hunk 8 above)
[the remark this answers is not in the document]
Comment (jimid27) [reply .github/workflows/release_build_deploy.yaml]: I'm going with it

**Review by jamiezieziula — APPROVED** [review]

**Pull request merged**